lightpath/ui/gui.py

```diff
@@ -411,3 +421,33 @@ def resizeEvent(self, evt):
     def closeEvent(self, a0) -> None:
         self._destroy_lightpath_summary_signals()
         return super().closeEvent(a0)
+
+
+class LoadingSplash(QDialog):
+    """ simple loading splash screen """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.setWindowFlags(Qt.Window | Qt.FramelessWindowHint)
+
+        self.setStyleSheet("QDialog { border: 2px solid; "
+                           "border-color:#00ffff }")
+        layout = QVBoxLayout()
+        self.setLayout(layout)
+
+        self.status_display = QLabel()
+        tout = typhos.utils.TyphosLoading.LOADING_TIMEOUT_MS
+        # No Timeout!
+        typhos.utils.TyphosLoading.LOADING_TIMEOUT_MS = -1
+        loading = typhos.utils.TyphosLoading(self)
+        typhos.utils.TyphosLoading.LOADING_TIMEOUT_MS = tout
+
+        status_layout = QHBoxLayout()
+        status_layout.addWidget(self.status_display)
+        status_layout.addWidget(loading)
+
+        layout.addLayout(status_layout)
+
+    def update_status(self, msg):
+        self.status_display.setText(f"Loading: {msg}")
+        # self.move()
+        QApplication.instance().processEvents()
```

Comment:
I think this is the only bit I don't understand- why do we need to explicitly process events here?

Reply:
If you don't process events, the QDialog will appear but not render any of the text or internals.  I'm not entirely sure why, but I think it has to do with processing the events in that thread, and not letting `change_path_display` block.

Reply:
Is this also why the animation doesn't play when we switch subdisplays?

Reply:
I believe so.  I haven't been able to find a way to get the animation to run, but I thought it might be ok to leave as is for now.

Reply:
I think it's as you guessed - you're tying up the GUI thread in `change_path_display` and not allowing Qt to process events for other events happening in the background (or even foreground redraws).

Reply:
Following a brilliant suggestion from zach, I just replaced the animation with a static loading icon.